src/topology/algebra/group.lean

```diff
@@ -1032,6 +1033,64 @@ begin
   exact topological_group.t1_space (G ⧸ S) ((quotient_map_quotient_mk.is_closed_preimage).mp hS),
 end
 
+/-- A discrete subgroup of a topological group `G` acts on `G` properly discontinuously on the left.
+-/
+@[to_additive "A discrete subgroup of an additive topological group `G` acts on `G` properly
+discontinuously on the left."]
+lemma subgroup.properly_discontinuous_smul_of_tendsto_cofinite
+  (S : subgroup G) (hS : tendsto S.subtype cofinite (cocompact G)) :
+  properly_discontinuous_smul S G :=
+{ finite_disjoint_inter_image := begin
+    intros K L hK hL,
+    have H : set.finite _ := hS ((hL.prod hK).image continuous_div').compl_mem_cocompact,
+    convert H,
+    ext x,
+    obtain ⟨x, hx⟩ := x,
+    dsimp,
+    simp_rw [set.ext_iff, prod.exists],
+    calc ¬(∀ l, (∃ k, k ∈ K ∧ x * k = l) ∧ l ∈ L ↔ false)
+        ↔ ¬¬∃ l k, (l ∈ L ∧ k ∈ K) ∧ x * k = l : by tidy
+    ... ↔ ¬¬∃ l k, (l ∈ L ∧ k ∈ K) ∧ l / k = x :
+            begin
+              congrm ¬¬∃ l k, _ ∧ _,
+              rw [div_eq_iff_eq_mul, @comm G (=)],
+            end
```

Comment:
I think it's worth polishing this a bit more.

How about something along the lines of:
```suggestion
    rw [preimage_compl, compl_compl] at H,
    convert H,
    ext ⟨x, hx⟩,
    suffices : x • K ∩ L ≠ ∅ ↔ x ∈ L / K, -- Warning this is `/` not `⧸` :-(
    { have h : (⟨x, hx⟩ : ↥S) • K = x • K, { ext y, refl, }, -- Missing `simp` lemma for `set.has_smul_set`?
      simp only [this, h, image_smul, subgroup.coe_subtype, subgroup.coe_mk, image_prod, image2_div], },
    -- Looks like missing lemma (worth breaking out).
```

Reply:
I liked this idea and I've tried it out (I can push the WIP if you think it would be helpful), but hit the following problem: for the "right-multiplication" version of the lemma, the best statement I came up with was
```
x • K ∩ L = ∅ ↔ unop x ∉ K⁻¹ * L
```
and `unop x ∉ K⁻¹ * L` is less well-behaved than `x ∈ L / K`: there are not really any lemmas about `K⁻¹ * L`, and it's constructed as an image of an image (`⁻¹`, `*`) rather than just an image.  I can't come up with any approach that keeps the symmetry between the two lemmas (which is transparent in the current version).

Let me know what you think.

Reply:
Good point about the symmetry (I hadn't even looked at the right version). Of course the source of asymmetry is the fact that we have this weird `has_div` class for `a * b⁻¹` but no such class for `a⁻¹ * b` and then the left proof uses this but the right proof does not. I'd argue that if you _really_ want symmetry, then the thing to do would be not to use `has_div` at all.

In any case, my main issue is that we should separate out the minor algebraic issues from this topological result. Here's a proposal:
```lean
@[to_additive] lemma group.smul_inter_ne_empty_iff {s t : set G} {x : G} :
  x • s ∩ t ≠ ∅ ↔ ∃ a b, (a ∈ t ∧ b ∈ s) ∧ a * b⁻¹ = x :=
begin
  rw ne_empty_iff_nonempty,
  split,
  { rintros ⟨a, h, ha⟩,
    obtain ⟨b, hb, rfl⟩ := mem_smul_set.mp h,
    exact ⟨x • b, b, ⟨ha, hb⟩, by simp⟩, },
  { rintros ⟨a, b, ⟨ha, hb⟩, rfl⟩,
    exact ⟨a, mem_inter (mem_smul_set.mpr ⟨b, hb, by simp⟩) ha⟩, },
end

-- Maybe we won't even use this if we opt for a fully symmetric approach.
@[to_additive] lemma group.smul_inter_ne_empty_iff' {s t : set G} {x : G} :
  x • s ∩ t ≠ ∅ ↔ ∃ a b, (a ∈ t ∧ b ∈ s) ∧ a / b = x :=
by simp_rw [group.smul_inter_ne_empty_iff, div_eq_mul_inv]

@[to_additive] lemma group.op_smul_inter_ne_empty_iff {s t : set G} {x : Gᵐᵒᵖ} :
  x • s ∩ t ≠ ∅ ↔ ∃ a b, (a ∈ s ∧ b ∈ t) ∧ a⁻¹ * b = unop x :=
begin
  sorry,
end

/-- A discrete subgroup of a topological group `G` acts on `G` properly discontinuously on the left.
-/
@[to_additive "A discrete subgroup of an additive topological group `G` acts on `G` properly
discontinuously on the left."]
lemma subgroup.properly_discontinuous_smul_of_tendsto_cofinite
  (S : subgroup G) (hS : tendsto S.subtype cofinite (cocompact G)) :
  properly_discontinuous_smul S G :=
{ finite_disjoint_inter_image := begin
    intros K L hK hL,
    have H : set.finite _ := hS ((hL.prod hK).image continuous_div').compl_mem_cocompact,
    rw [preimage_compl, compl_compl] at H,
    convert H,
    ext ⟨x, hx⟩,
    simpa only [image_smul, mem_image, prod.exists] using group.smul_inter_ne_empty_iff',
  end }

local attribute [semireducible] mul_opposite

/-- A discrete subgroup of a topological group `G` acts on `G` properly discontinuously on the
right.

If `G` is Hausdorff, this can be combined with `t2_space_of_properly_discontinuous_smul_of_t2_space`
to show that the quotient group `G ⧸ S` is Hausdorff. -/
@[to_additive "A discrete subgroup of an additive topological group `G` acts on `G` properly
discontinuously on the right.

If `G` is Hausdorff, this can be combined with `t2_space_of_properly_discontinuous_vadd_of_t2_space`
to show that the quotient group `G ⧸ S` is Hausdorff."]
lemma subgroup.properly_discontinuous_smul_opposite_of_tendsto_cofinite
  (S : subgroup G) (hS : tendsto S.subtype cofinite (cocompact G)) :
  properly_discontinuous_smul S.opposite G :=
{ finite_disjoint_inter_image := begin
    intros K L hK hL,
    have : continuous (λ p : G × G, (p.1⁻¹, p.2)) := continuous_inv.prod_map continuous_id,
    have H : set.finite _ :=
      hS ((hK.prod hL).image (continuous_mul.comp this)).compl_mem_cocompact,
    rw [preimage_compl, compl_compl] at H,
    convert H,
    ext ⟨x, hx⟩,
    simpa only [image_smul, mem_image, prod.exists] using group.op_smul_inter_ne_empty_iff,
  end }
```

Reply:
Nice, I like it!